include/aws/common/array_list_with_map.h

```diff
@@ -0,0 +1,82 @@
+#ifndef AWS_COMMON_ARRAY_LIST_WITH_MAP_H
+#define AWS_COMMON_ARRAY_LIST_WITH_MAP_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/array_list.h>
+#include <aws/common/common.h>
+#include <aws/common/hash_table.h>
+
+struct aws_array_list_with_map {
```

Comment:
Maybe some explanation on the relationship between the map and the list? If I understand GfG:
list is a list of elements.
map is a map whose keys are elements, and value are array indexes.

Reply:
I think two lines after that explains this a bit